src/realm-settings/LocalizationTab.tsx

```diff
@@ -76,7 +81,18 @@ export const LocalizationTab = ({
     if (realm) {
       const result = await adminClient.realms.getRealmLocalizationTexts({
         realm: realm.realm!,
-        selectedLocale: getValues("defaultLocale") || "en",
+        selectedLocale: selectMenuLocale || getValues("defaultLocale") || "en",
+      });
+      return Object.keys(result).map((key) => [key, result[key]]);
+    }
+    return [[]];
+  };
+
+  const tableLoader = async () => {
+    if (realm) {
+      const result = await adminClient.realms.getRealmLocalizationTexts({
+        realm: currentRealm,
+        selectedLocale: selectMenuLocale,
       });
       return Object.keys(result).map((key) => [key, result[key]]);
```

Comment:
Not entirely sure, but we might be able to replace this with `Object.entries()`.
```suggestion
      return Object.entries(result);
```

Reply:
Thank you for the suggestion! Looks like the loader prop on KeycloakDataTable doesn't like this return type